.github/workflows/test.yml

```diff
@@ -16,13 +16,23 @@ permissions:
 jobs:
   build:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        toolchain:
+          - "1.70"
+          - "stable"
+          - "beta"
+          - "nightly"
 
     steps:
-    - name: Checkout repository
-      uses: actions/checkout@v4
+      - name: Checkout repository
+        uses: actions/checkout@v4
 
-    - name: Build
-      run: cargo build --verbose
+      - name: Update Rust toolchain
+        run: rustup update ${{ matrix.toolchain }} && rustup default ${{ matrix.toolchain }}
```

Comment:
use https://github.com/dtolnay/rust-toolchain?

Reply:
I'm slightly paranoid with relying on 3rd party GitHub Actions after the recent `tj-actions/changed-files` compromise, probably ok to use the built-in `rustup` here for now.